src/fpmath.h

```diff
@@ -47,7 +47,7 @@
 #define _PDP_ENDIAN       __ORDER_PDP_ENDIAN__
 #define _BYTE_ORDER       __BYTE_ORDER__
 
-#elif defined(__linux)
+#if defined(__linux__) || defined(__FreeBSD_kernel__) || defined(__GNU__)
```

Comment:
If I understand the code correctly, we want to test whether the system uses glibc. Instead, we now test whether the system has a Linux kernel, a FreeBSD kernel or a HURD kernel. If your system has a FreeBSD kernel, it may not necessarily have the headers you're trying to include. FreeBSD itself does not have these headers.